Stellar-ledger-entries.x

```diff
@@ -493,58 +494,32 @@ struct LiquidityPoolEntry
     body;
 };
 
-enum ContractEntryBodyType {
-    DATA_ENTRY = 0,
-    EXPIRATION_EXTENSION = 1
-};
-
-const MASK_CONTRACT_DATA_FLAGS_V20 = 0x1;
-
-enum ContractDataFlags {
-    // When set, the given entry does not recieve automatic expiration bumps
-    // on access. Note that entries can still be bumped manually via the footprint.
-    NO_AUTOBUMP = 0x1
-};
-
 enum ContractDataDurability {
     TEMPORARY = 0,
     PERSISTENT = 1
 };
 
 struct ContractDataEntry {
+    ExtensionPoint ext;
+
     SCAddress contract;
     SCVal key;
     ContractDataDurability durability;
-
-    union switch (ContractEntryBodyType bodyType)
-    {
-    case DATA_ENTRY:
-    struct
-    {
-        uint32 flags;
-        SCVal val;
-    } data;
-    case EXPIRATION_EXTENSION:
-        void;
-    } body;
-
-    uint32 expirationLedgerSeq;
+    SCVal val;
 };
 
 struct ContractCodeEntry {
     ExtensionPoint ext;
 
     Hash hash;
-    union switch (ContractEntryBodyType bodyType)
-    {
-    case DATA_ENTRY:
-        opaque code<>;
-    case EXPIRATION_EXTENSION:
-        void;
-    } body;
+    opaque code<>;
+};
 
+struct ExpirationEntry {
+    // Hash of the LedgerKey that is associated with this ExpirationEntry
+    Hash keyHash;
```

Comment:
wait, isn't the entry supposed to hold the expiration sequence number, while the key holds the hash?

Reply:
A `LedgerEntry` must contain its `LedgerKey`. We don't store keys and values separately in the BucketList, we only store `LedgerEntry` and cast the `LedgerEntry` to a `LedgerKey` when searching by key.

Reply:
Oh okay. I think I was confused because I missed subtle little L521 due to the GH comments (right below this thread), so I was confused where the `expirationLedgerSeq` was stored now